kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -198,9 +199,20 @@ public Optional<SetTransaction> getSetTxnOpt() {
         return setTxnOpt;
     }
 
+    private Optional<Long> generateInCommitTimestampForFirstCommitAttempt(long currentTimestamp) {
+        if (IN_COMMIT_TIMESTAMPS_ENABLED.fromMetadata(metadata)) {
+            return Optional.of(currentTimestamp);
+        } else {
+            return Optional.empty();
+        }
+    }
+
     private Row generateCommitAction() {
+        long commitAttemptStartTime = System.currentTimeMillis();
```

Comment:
why are we not reading the existing timestamp, so that the next commit timestamp is monotonically increasing?